### PR TITLE
Binary insertion sort

### LINKED ISSUE #1806: Binary Insertion Sort in C#
@singh-shreya6 
This code isn't available in the repo. I would like to add this using C#. 
Please assign me the same under GSSoC'20.

### PR DESCRIPTION
**Pull request by Yashank18**
Fixes #1806 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added implementation of Binary Insertion Sort 

#### Languages Used:
- C#

#### Files Added:
- Binary_insertion-sort.cs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+class BIS { 
+    public static void sort(int []array) 
+    { 
+        for (int i = 1; i < array.Length; i++) 
+        { 
+            int x = array[i]; 
+  
+            // Find location to insert using 
+            // binary search 
+            int j = Math.Abs(Array.BinarySearch( array, 0, i, x) + 1);
```
Comment (SakshiShreya) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: Please remove the space before 'array'.

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            // binary search 
+            int j = Math.Abs(Array.BinarySearch( array, 0, i, x) + 1); 
+  
+            // Shifting array to one location right 
+            System.Array.Copy(array, j, array, j+1, i-j); 
+ 
+            array[j] = x; 
+        } 
+    }
+    public static void Main()
```
Comment (SakshiShreya) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: There should be 1 space between 2 functions.

File: Binary_Insertion_Sort/Binary_Insertion_Sort.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    public static void Main() 
+    { 
+        int size;
+        size= Convert.ToInt32(Console.ReadLine()); 
+
+        int[] array = new int[size];
+
+        for(int i=0; i<size; i++)
+	    {
```
Comment (SakshiShreya) [inline Binary_Insertion_Sort/Binary_Insertion_Sort.cs]: Please follow indentation properly. Use 4 spaces instead of 1 tab.

**Comment by Yashank18** [comment]
@SakshiShreya Updated!

**Pull request closed without merging**